src/cart/crt_iv.c

```diff
@@ -1320,6 +1388,16 @@ crt_hdlr_iv_fetch_aux(void *arg)
 	 */
 	IVNS_DECREF(ivns_internal);
 
+	/* Check group version match */
+	grp_ver_entry = ivns_internal->cii_grp_priv->gp_membs_ver;
+
+	if (grp_ver_entry != input->ifi_grp_ver) {
+		D_ERROR("Group (%s) version mismatch. Local: %d Remote :%d\n",
+			ivns_id.ii_group_name, grp_ver_entry,
+			input->ifi_grp_ver);
+		D_GOTO(send_error, rc = -DER_GRPVER);
+	}
```

Comment:
Is this check needed if it is done already after parent retrieval on line 1468? Why not just check input->ifi_grp_ver to current version in line 1468 check instead of checking here and then checking on line 1468 that it didnt change?

Reply:
The check at this point will eliminate the response if the group version has changed prior to this point, and will apply to both bulk and non-bulk transfer.  The test at  line 1468 only applies non-bulk transfer and will pick up any asynchronous changes between the 2 tests.

It is not just a matter of removing one.  The question is whether the group version check should apply to bulk transfer?

I'm also assuming that the earlier we can detect a version change and stop it from setting up other call backs, the better.